specs/DOMContentLoaded.md

```diff
@@ -0,0 +1,117 @@
+# Background
+
+In response to consumers' requests for an event similar to the old [WebView DOMContentLoaded](https://docs.microsoft.com/en-us/microsoft-edge/hosting/webview#mswebviewdomcontentloaded), the WebView2 team has introduced DOMContentLoaded API which indicates that the main DOM elements have finished loading.
+In this document we describe the new API. We'd appreciate your feedback.
+
+# Description
+We propose adding DOMContentLoaded to CoreWebView2. This allows the developer to have an event that fires when the DOM is loaded after the WebView2 navigates to a page.
+
+# Examples
+## Win32 C++
+```
+ScenarioDOMContentLoaded::ScenarioDOMContentLoaded(AppWindow* appWindow)
+    : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+{
+    //! [DOMContentLoaded]
+    // Register a handler for the DOMContentLoaded event.
+    // Check whether the DOM content loaded
+    CHECK_FAILURE(m_webView->add_DOMContentLoaded(
+        Callback<ICoreWebView2DOMContentLoadedEventHandler>(
+            [this](ICoreWebView2* sender, ICoreWebView2DOMContentLoadedEventArgs* args)
+                -> HRESULT {
+                m_webView->ExecuteScript(
+                    L"let "
+                    L"content=document.createElement(\"h2\");content.style.color='blue';"
+                    L"content.textContent=\"This text was added by the host "
+                    L"app\";document.body.appendChild(content);",
+                    Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
+                        [](HRESULT error, PCWSTR result) -> HRESULT { return S_OK; })
+                        .Get());
+                return S_OK;
+            })
+            .Get(),
+        &m_DOMContentLoadedToken));
+    //! [DOMContentLoaded]
+```
+
+## C#
+```
+webView.CoreWebView2.DOMContentLoaded += (object sender, CoreWebView2DOMContentLoadedEventArgs arg) =>
+{
+    webView.ExecuteScriptAsync("let "
+                              "content=document.createElement(\"h2\");content.style.color="
+                              "'blue';content.textContent= \"This text was added by the "
+                              "host app\";document.body.appendChild(content);");
+};
+webView.NavigateToString(@"<!DOCTYPE html><h1>DOMContentLoaded sample page</h1><h2>The content below will be added after DOM content is loaded </h2>");
+
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2_2;
+interface ICoreWebView2DOMContentLoadedEventArgs;
+interface ICoreWebView2DOMContentLoadedEventHandler;
+
+[uuid(9810c82b-8483-4f1c-b2f4-6244f1010c05), object, pointer_default(unique)]
+interface ICoreWebView2_2 : ICoreWebView2 {
+  /// Add an event handler for the DOMContentLoaded event.
+  /// DOMContentLoaded fires when the initial html document has been parsed.
```

Comment:
Is it okay to say that the event "fires"?

Windows Runtime convention is to say that events are "raised". HTML DOM specification use the term "[dispatched](https://dom.spec.whatwg.org/#concept-event-dispatch)".